modules/apps/asset/asset-categories-admin-web/src/main/java/com/liferay/asset/categories/admin/web/internal/portlet/action/EditAssetVocabularyMVCActionCommand.java

```diff
@@ -177,8 +177,20 @@ else if (Objects.equals(required, "required")) {
 			}
 		}
 
-		AssetVocabularySettingsHelper vocabularySettingsHelper =
-			new AssetVocabularySettingsHelper();
+		long vocabularyId = ParamUtil.getLong(actionRequest, "vocabularyId");
+
+		AssetVocabulary assetVocabulary = _assetVocabularyService.fetchVocabulary(
```

Comment:
This was the issue. I used a get method instead of a fetch which failed on add as there is nothing to get yet.

Reply:
Good catch